tubular/dates.py

```diff
@@ -14,132 +13,196 @@
 
 class BaseDateTransformer(BaseTransformer):
     """
-    Transformer with date data checks needed by all transformers taking dates as input data.
+    Extends BaseTransformer for datetime scenarios.
+
+    Parameters
+    ----------
+    columns : List[str]
+        List of 2 columns. First column will be subtracted from second.
+
+    new_column_name : str
+        Name for the new year column.
+
+    drop_original : bool
+        Flag for whether to drop the original columns.
+
+    **kwargs
+        Arbitrary keyword arguments passed onto BaseTransformer.init method.
+
+    Attributes
+    ----------
+    columns : List[str]
+        List of 2 columns. First column will be subtracted from second.
+
+    new_column_name : str
+        Name for the new year column.
+
+    drop_original : bool
+        Flag for whether to drop the original columns.
+
     """
 
-    def _generate_is_datetime64_dict(self, X: pd.DataFrame) -> None:
-        """
-        Function to generate a dictionary attribute storing the result of pd.api.types.is_datetime64_any_dtype()
-        for each column in self.columns to avoid repeated calls to this function.
-        """
+    def __init__(
+        self,
+        columns: list,
+        new_column_name: str,
+        drop_original: bool = False,
+        **kwargs: dict[str, bool],
+    ) -> None:
+        super().__init__(columns=columns, **kwargs)
 
-        self._is_datetime64_dict = {}
+        if not (isinstance(new_column_name, str)):
+            msg = f"{self.classname()}: new_column_name should be str"
+            raise TypeError(msg)
 
-        for col in self.columns:
-            self._is_datetime64_dict[col] = pd.api.types.is_datetime64_any_dtype(X[col])
+        if not (isinstance(drop_original, bool)):
```

Comment:
S - imminent incoming change in #248  may allow inheriting this bit of logic.  Worth small change if #248 in before this one